src/main/java/com/cloudant/client/api/Database.java

```diff
@@ -245,6 +246,9 @@ public void createIndex(String indexDefinition) {
 				list.add(t);
 			}
 			return list;
+		 } catch (UnsupportedEncodingException e) {
+			// This should never happen as every implementation of the java platform is required to support UTF-8.
+			return null;

```

Comment:
On balance I think it's probably a good idea. I was going on the basis that as it should never happen it doesn't really matter. However, it's probably a bit cleaner to throw a RuntimeException and would give us a chance of identifying the issue from a stack trace if it ever did happen on a non-compliant/buggy platform.